src/log.rs

```diff
@@ -28,15 +38,29 @@ pub fn show(mut connection: Connection, verbose: bool) {
                 } else {
                     Time::Local(Local::now().naive_local())
                 };
-                let qos = format::qos(publish.qos);
                 let topic = publish.topic;
                 let size = publish.payload.len();
                 let payload = Payload::unlimited(publish.payload.into());
-                println!("{time:12} QoS:{qos:11} {topic:50} Payload({size:>3}): {payload}");
+                if json {
+                    println!(
+                        "{}",
+                        serde_json::to_string(&JsonLog {
```

Comment:
Personally I would move this out of the println macro. Code in macros has not as good language server tooling as macros are manipulating the code themselves.

Also I think this would save 2 lines as the println wouldn't be multiline anymore.